tutorial_plugin_java.md

```diff
@@ -0,0 +1,154 @@
+ Example:
+ 
+
+        {
+        
+    "fields": [ 
+    {
+    
+    "type": "ObjectIdType",
+    
+    "logical_type": "OID",
+    
+    "field_name": "_id",
+    
+    "desc": "Identifier of the document."
+    
+    },
+    
+    {
+    
+    "sub_type": "StructType",
+    
+    "type": "ArrayType",
+    
+    "field_name": "induces",
+    
+    "logical_type": "List",
+    
+    "desc": "List of file actions this file action induces.",
+    
+    "fields": [
+    
+    {
+    
+    "desc": "Reference to the file action this file action induces.",
+    
+    "field_name": "change_file_action_id",
+    
+    "logical_type": "RID",
+    
+    "reference_to": "file_action",
+    
+    "type": "ObjectIdType"
+    
+    },
+    
+    {
+    
+    "type": "StringType",
+    
+    "logical_type": "Name",
+    
+    "field_name": "szz_type",
+    
+    "desc": "SZZ type for this inducing file action as per the original algorithm (weak_suspect, hard_suspect, partial_fix, inducing)"
+    
+    },
+    
+    {
+    
+    "type": "StringType",
+    
+    "logical_type": "Name",
+    
+    "field_name": "label",
+    
+    "desc": "SmartSHARK label for this inducing file action."
+    
+    }
+    
+    ]
+    
+    }
+    
+    ],
+    
+    "desc": "",
+    
+    "collection_name": "file_action"
+    
+    }
+
+ ### Build and Installation
+ Execute the following commands to build the plugin and create an archive for uploading to the SmartShark Server.
+ 
+    ./gradlew build
+    ./plugin-packaging/build_plugin.sh
+
+### Add external libraries
+You are able to add any external library to the build.gradle file in the root folder.
+Additional information can be found in the documentation of gradle [here](https://gradle.org/).
+
+## Files
+In this section, we will provide detailed information about any file of the minimal working example. 
+### .travis.yml
+File for the Travis-Build System. 
+### JmweSHARK.java
+The java main class of the plugin. The main method is automatically called by execution of the plugin.
+### build.gradle
+Additional information about the build.gradle file can be found in the documentation from gradle [here](https://docs.gradle.org/current/userguide/tutorial_using_tasks.html).
+### build_plugin.sh
+Script for creating a zip archive file of the plugin that is able to upload to the ServerSHARK. 
+### execute.sh
+The plugin is execution starts with this script file. You may add here some additional parameters for your plugin.
+### info.json
+Contains important meta information about the plugin. This file is readed by the ServerSHARK to ensure that the requirements to execute the plugin are given. 
+### install.sh
+Install-Script: Usually, you do not need to change anything here. 
+### schema.json
+Contains any information about database changes that are possible while executing this plugin.
```

Comment:
"Contains information about collections and fields that may be modified in the database by this plugin."